grails-app/views/occurrence/_map.gsp

```diff
@@ -1145,6 +1207,79 @@
       delete link;
       return false;
     }
+
+    function updateWmsModalContent() {
+        // Get current WMS layer parameters
+        var currentLayer = MAP_VAR.currentLayers[0];
+        if (currentLayer) {
+            var wmsParams = currentLayer.wmsParams;
+            var baseUrl = MAP_VAR.mappingUrl + "/mapping/wms/reflect";
+
+            // Format WMS parameters in tabular format
+            var formattedParams = Object.keys(wmsParams).map(function(key) {
+                // Pad parameter name to 20 chars for alignment
+                var paddedKey = (key.toUpperCase() + ':').padEnd(20, ' ');
+                return paddedKey + wmsParams[key];
+            }).sort().join('\n');
+
+
+            // Remove any existing dynamic keys from formattedParams
+            formattedParams = formattedParams.split('\n').filter(line =>
+                !line.startsWith('SIZE:') &&
+                !line.startsWith('STYLE:') &&
+                !line.startsWith('OUTLINE:') &&
+                !line.startsWith('COLOUR:')
+            ).join('\n');
+
+            // Add dynamic values to WMS parameters (current map tool values)
+            formattedParams += '\n' + 'SIZE:'.padEnd(20, ' ') + $('#sizeslider-val').html();
+            formattedParams += '\n' + 'STYLE:'.padEnd(20, ' ') + "opacity:" + $('#opacityslider-val').html(); // for grid data
+            formattedParams += '\n' + 'OUTLINE:'.padEnd(20, ' ') + $('#outlineDots').is(':checked');
+            formattedParams += '\n' + 'COLOUR:'.padEnd(20, ' ') + $('#pcolour').val().replace('#','').toUpperCase();
+
+            // query is stored in MAP_VAR.query and fq is stored in MAP_VAR.removeFqs ?
+            formattedParams += '\n' + 'q:'.padEnd(20, ' ') + MAP_VAR.query;
+            formattedParams += '\n' + 'fq:'.padEnd(20, ' ') + "-occurrence_status:absent";
+
+            // Construct full URL
+            var fullUrl = baseUrl + '?' + Object.keys(wmsParams).map(function(key) {
+                return key + '=' + wmsParams[key];
+            }).join('&');
+
+            // Update modal fields
+            $('#wmsBaseUrl').val(baseUrl);
+            $('#wmsParams').val(formattedParams);
+            $('#wmsFullUrl').val(fullUrl);
+        }
+    }
+
+    // Add event listeners to map controls that trigger WMS updates
```

Comment:
I've no idea, but my first thought was why do you need these listeners because can't you just update the modal content when it opens?

Reply:
Your right the listeners are not required. I'll remove them.